portal/app/algorithms/battery_load_control.py

```diff
@@ -0,0 +1,53 @@
+import json, datetime, numpy as np
+from app.farm_updater import sonnen_api
+from django.conf import settings
+
+
+def batt_dispatch():
+    from app.models import FarmDevice, FarmData, DeviceType
+
+    egauge_device = FarmDevice.objects.get(device_uid=settings.EGAUGE_ID)
+    sonnen_queryset = FarmDevice.objects.filter(type=DeviceType.SONNEN)
+    farmdata = FarmData.objects.filter(farm_device=egauge_device).order_by('-id')[0:3]
+    batt_instance = sonnen_api.SonnenApiInterface()
+    test_pen_power = []
+
+    # Getting egauge info
+    for data in farmdata:
+        test_pen_power.append(json.loads(data.device_data)['processed']['POWER_CIRCUIT1'] + json.loads(data.device_data)['processed']['POWER_CIRCUIT2'])
+
+    avg_test_pen_power = np.average(np.asarray(test_pen_power))
+
+    # getting batt soc
```

Comment:
You should convert everything below this comment into a loop based on the two entries in the queryset. You are basically repeating a ton of code here. You can simplify it.

Reply:
I've changed it but I'll miss one point that I wanted to have to make the system more electrically unstable. I'm pretty sure there is a clever way to do it but for now I think is ok